### ISSUE TITLE
Question about low performance in the fMRI modality

### ISSUE THREAD

**Issue by AIyumeng** [ncclab-sustech/BrainFLORA#2]
Hello, thank you very much for sharing your code and data.
While reproducing the results from the paper, I encountered a few questions:

Using the same settings, I was able to achieve results similar to those reported in the paper for the EEG and MEG modalities.
However, the fMRI performance is noticeably lower than what was reported.
Could you please clarify what might be causing this discrepancy?

Additionally, I noticed that MEG subject sub02 shows significantly better performance than other subjects.
Is this expected behavior? Was there anything special about this subject mentioned in the paper or dataset documentation?

I would really appreciate your insights and guidance on these points.
Thank you for your time and for sharing your great work!

<img width="1780" height="736" alt="Image" src="https://github.com/user-attachments/assets/944f6e36-dbac-4b6f-b583-14f06fb595e9" />

**Comment by KevinZhou6** [comment]
I encountered this problem as well. May I ask what your solution was?

**Comment by AIyumeng** [comment]
No progress.

**Comment by dongyangli-del** [comment]
Hi @AIyumeng, thank you for your detailed feedback and for your effort in reproducing our work! I am glad to hear that you achieved consistent results on the EEG and MEG modalities.

1. fMRI Performance Discrepancy: fMRI performance is highly sensitive to data preprocessing steps and feature alignment. To resolve this discrepancy and ensure you are working with the exact same baseline, I have recently updated the repository and released the preprocessed datasets and pretrained weights on Hugging Face. I strongly recommend running the inference using these files to verify the results.

🔗 Hugging Face: https://huggingface.co/datasets/LidongYang/BrainFLORA

2. MEG Subject sub02 Performance: Yes, this is expected behavior. In neuroimaging studies, inter-subject variability is quite common. We also observed that sub02 exhibits significantly higher Signal-to-Noise Ratio (SNR) and better neural responses compared to other subjects, leading to superior decoding performance.

Please feel free to try the new weights and let me know if the fMRI results improve.

**Comment by AIyumeng** [comment]
@dongyangli-del  Thank you for your reply.

I attempted to use the updated code and newly open-sourced models. Encouragingly, I was able to successfully reproduce the fMRI modality results using the provided checkpoints. However, I was unable to reproduce the training of the fMRI model and discovered several new issues.

First, I evaluated the fMRI performance using eval/FLORA_inference.ipynb and obtained relatively reliable experimental results.

| Subject | Acc   | Top-5 | k=2  | k=4  | k=10 |
|--------|-------|-------|------|------|------|
| sub-01 | 0.2500 | 0.5400 | 0.9200 | 0.8200 | 0.6000 |
| sub-02 | 0.2000 | 0.5900 | 0.9100 | 0.8000 | 0.6700 |
| sub-03 | 0.1700 | 0.4500 | 0.9100 | 0.8100 | 0.5400 |
| **Avg** | **0.2067** | **0.5267** | **0.9133** | **0.8100** | **0.6033** |

However, I was not able to reproduce the results using the weights provided in features. I also noticed that the img_features of some modalities were not normalized as expected. Is this behavior intended?

I trained the model using the unnormalized features, but even after multiple epochs the performance remained close to the chance level.  I manually interrupted the experiment and then proceeded to use the normalized features
```
    img_features_train_all /= img_features_train_all.norm(dim=-1, keepdim=True)
    img_features_test_all /= img_features_test_all.norm(dim=-1, keepdim=True)
```
The results I obtained were still similar to my previous ones and did not reach the performance of the open-sourced weights.

**Comment by AIyumeng** [comment]
In a second update, the poor fMRI performance is more likely due to the padding in the Retrieval fMRIdataset. Changing the padding from 11,000 to 7,000 improved the performance. As for the open-sourced feature weights not being normalized, I have no further comments; after applying normalization, I obtained better results.
